### PR TITLE
Adding all models

### LINKED ISSUE #33: Create all models according to the class diagram

### PR DESCRIPTION
**Pull request by felipeaugustosm**
Resolves #33

### CODE REVIEW

**Comment by lucasheriques** [comment]
@felipeaugustosm can you please generate a model using MySQL Workbench? I think its better to check if we see using it.

File: code/restful-api/restful-api/Models/Cria.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using System;
+namespace RestfulApi.Models
+{
+    public class Cria
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Cria.cs]: A better name for this should be `OrganizadorUsuario`, no?

`Cria` is very vague.

File: code/restful-api/restful-api/Models/Historico_Eventos.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace RestfulApi.Models
+{
+    public class Historico_Eventos
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Historico_Eventos.cs]: I think we don't need this class, since it's just going to be a consult on the DB like 'give me all events that this user with id X came'. What do you think?

File: code/restful-api/restful-api/Models/Ingresso.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        public string Nome { get; set; }
+
+        public string Sobrenome { get; set; }
+
+        public string Celular { get; set; }
+
+        public string Email { get; set; }
+
+        public Historico_Eventos Historico_EventosId { get; set; }
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Ingresso.cs]: As I said previously, I don't think this is needed.

File: code/restful-api/restful-api/Models/InstituicaoFinanceira.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+using System;
+
+namespace RestfulApi.Models
+{
+    public class InstituicaoFinanceira
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/InstituicaoFinanceira.cs]: Is this class related to anything? Could not find it.

File: code/restful-api/restful-api/Models/Local.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+using System;
+namespace RestfulApi.Models
+{
+    public class Local
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Local.cs]: I think it makes more sense if `Local` has an `Endereco`, and not otherwise.

What do you think?

File: code/restful-api/restful-api/Models/Participa.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RestfulApi.Models
+{
+    public class Participa
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Participa.cs]: Maybe we should name this `UsuarioSorteio` if we decide to go with this naming approach.

File: code/restful-api/restful-api/Models/Possui.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RestfulApi.Models
+{
+    public class Possui
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Possui.cs]: Same for above, if we go with that naming approach, this should be `UsuarioCartaoDeCredito`.

File: code/restful-api/restful-api/Models/Possui.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        public long Id { get; set; }
+        
+        CartaoDeCredito CartaoDeCredito { get; set; }
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Possui.cs]: Missing `public` here, and also the Id.

File: code/restful-api/restful-api/Models/Possui.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+        Usuario Usuario { get; set; }
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Possui.cs]: Missing `public` here, and also the Id.

File: code/restful-api/restful-api/Models/Promocao.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
+using System;
+namespace RestfulApi.Models
+{
+    public class Promocao
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Promocao.cs]: I think this should probably be linked with `Evento` instead of `Organizador`.

Why did you decide to go with `Organizador` instead?

File: code/restful-api/restful-api/Models/Realiza.cs [hunk 11]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RestfulApi.Models
+{
+    public class Realiza
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Realiza.cs]: Why is this class needed? Since it is an 1-1 relationship, as one `Compra` is paid with only one `CartaoDeCredito`.

File: code/restful-api/restful-api/Models/RedesSociais.cs [hunk 12]
```diff
@@ -0,0 +1,5 @@
+using System;
+public class RedesSociais
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/RedesSociais.cs]: Why do we have `IRedesSociais` and this?

File: code/restful-api/restful-api/Models/Sorteio.cs [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+        public string Premios { get; set; }
+
+        public string Observacao { get; set; }
+
+        public long OrganizadorId { get; set; }
+
+        public Organizador Organizador { get; set; }
+
+        public Usuario[] Usuario { get; set; }
```
Comment (lucasheriques) [inline code/restful-api/restful-api/Models/Sorteio.cs]: Did you check if this is the proper format for a list of `Usuario`? I'm not sure.

File: code/restful-api/restful-api/Controllers/EventoesController.cs [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using RestfulApi.Models;
+
+namespace RestfulApi.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/Eventoes")]
+    public class EventoesController : Controller
```
[the remark this answers is not in the document]
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Controllers/EventoesController.cs]: Alterado para EventosController

File: code/restful-api/restful-api/Models/CartaoDeCredito.cs [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+        public long Id { get; set; }
+        public string Bandeira { get; set; }
+
+        public string NumeroDoCartao { get; set; }
+
+        public string NomeImpressoCartao { get; set; }
+
+        public DateTime Validade { get; set; }
+
+        public char CodigoDeSeguranca { get; set; }
```
[the remark this answers is not in the document]
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/CartaoDeCredito.cs]: Alterado para int

File: code/restful-api/restful-api/Models/CartaoDeCredito.cs [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+
+        public long CompraID { get; set; }
```
[the remark this answers is not in the document]
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/CartaoDeCredito.cs]: Criado CartaoDeCreditoCompra

File: code/restful-api/restful-api/Models/Compra.cs [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+namespace RestfulApi.Models
+{
+
+    public class Compra : IPaypal
+    {
+        public long Id { get; set; }
+
+        public float ValorCompra { get; set; }
+
+        public int QuantidadeIngressos { get; set; }
```
[the remark this answers is not in the document]
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Compra.cs]: Criado ItensCompra

File: code/restful-api/restful-api/Models/Compra.cs [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+
+        public DateTime DataCompra { get; set; }
+
+        public DateTime DataVencimento { get; set; }
+
+        public string FormaPagamento { get; set; }
+
+        public string Status { get; set; }
+
+        public long IngressoId { get; set; }
```
[the remark this answers is not in the document]
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Compra.cs]: Criado ItensCompra

File: code/restful-api/restful-api/Models/Possui.cs (hunk 9 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Possui.cs]: Coloquei public nas duas variáveis

File: code/restful-api/restful-api/Models/Promocao.cs (hunk 10 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Promocao.cs]: Removido o Organizador e criado o Eventos

File: code/restful-api/restful-api/Models/Realiza.cs (hunk 11 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Realiza.cs]: Excluido a class Realiza

File: code/restful-api/restful-api/Models/RedesSociais.cs (hunk 12 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/RedesSociais.cs]: Retirada a classe RedeSociais

File: code/restful-api/restful-api/Models/Sorteio.cs (hunk 13 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Sorteio.cs]: Criado a class SorteioGanhadores para realizar relacionamento n-n. Retirado usuarios.

File: code/restful-api/restful-api/Controllers/EventoesController.cs (hunk 14 above)
[the remark this answers is not in the document]
Comment (lucasheriques) [reply code/restful-api/restful-api/Controllers/EventoesController.cs]: 👍

File: code/restful-api/restful-api/Models/Compra.cs (hunk 17 above)
[the remark this answers is not in the document]
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/Compra.cs]: 👍

File: code/restful-api/restful-api/Models/Possui.cs (hunk 9 above)
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/Possui.cs]: 👍

File: code/restful-api/restful-api/Models/Promocao.cs (hunk 10 above)
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/Promocao.cs]: 🆗

File: code/restful-api/restful-api/Models/Realiza.cs (hunk 11 above)
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/Realiza.cs]: 🆗

File: code/restful-api/restful-api/Models/RedesSociais.cs (hunk 12 above)
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/RedesSociais.cs]: 🆗

File: code/restful-api/restful-api/Models/Sorteio.cs (hunk 13 above)
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/Sorteio.cs]: 👍

File: code/restful-api/restful-api/Models/Contem.cs [hunk 19]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RestfulApi.Models
+{
+    public class Contem
```
[the remark this answers is not in the document]
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Contem.cs]: Retirado do projeto

File: code/restful-api/restful-api/Models/Historico_Eventos.cs (hunk 2 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Historico_Eventos.cs]: Retirado.

File: code/restful-api/restful-api/Models/Ingresso.cs (hunk 3 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Ingresso.cs]: Retirado.

File: code/restful-api/restful-api/Models/InstituicaoFinanceira.cs (hunk 4 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/InstituicaoFinanceira.cs]: Retirado. Substituido pelo IPayPal

File: code/restful-api/restful-api/Models/Local.cs (hunk 5 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Local.cs]: Feito.

File: code/restful-api/restful-api/Models/Participa.cs (hunk 6 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Participa.cs]: Feito.

File: code/restful-api/restful-api/Models/Possui.cs (hunk 7 above)
Comment (felipeaugustosm) [reply code/restful-api/restful-api/Models/Possui.cs]: Feito

File: code/restful-api/restful-api/Models/CartaoDeCredito.cs (hunk 16 above)
[the remark this answers is not in the document]
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/CartaoDeCredito.cs]: ok

File: code/restful-api/restful-api/Models/Compra.cs (hunk 18 above)
[the remark this answers is not in the document]
Comment (lucasheriques) [reply code/restful-api/restful-api/Models/Compra.cs]: ok
